### PR TITLE
Feature/hdf 13 APY

### PR DESCRIPTION
**Pull request by barryhalo**
## Description of Changes
- APY calculation [Formula](https://halodao.atlassian.net/browse/HDF-13?focusedCommentId=11733)
- Currently using ETH price
- apy = (monthlyInterest * 100) * 12
- Unit tests will be on new task/branch
- Still need some refactoring

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-13)

### Todos:
- Coingecko Halo price as dummy price

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [x] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/components/Farm/FarmPoolCard.tsx [hunk 1]
```diff
@@ -403,22 +404,46 @@ export default function FarmPoolCard({ poolInfo, tokenPrice }: FarmPoolCardProps
...
+  const rewardTokenPerSecond = useRewardTokenPerSecond();
+
+  // 1 month in seconds
+  // (days * hrs * min * s) * reward token per second
+  const monthlyReward = rewardTokenPerSecond ? (30 * 24 * 60 * 60) * rewardTokenPerSecond : 0;
+  const totalAllocPoint = useTotalAllocPoint();
+  const WETHAddr = poolInfo.tokens[1].address;  // WETH address
+  const USDPrice = tokenPrice[WETHAddr];
+  const rewardMonthUSDValue = (poolInfo.allocPoint / totalAllocPoint) * (monthlyReward * USDPrice);
+  const monthlyAPY = rewardMonthUSDValue / poolLiquidity;
```
Comment (barryhalo) [inline src/components/Farm/FarmPoolCard.tsx]: @ctverceles should I rename `monthlyAPY` to `monthlyInterest` since this is not in percentage?

**Review by tracyarciaga — COMMENTED** [review]
@barryhalo can we format this APY column to include a comma? (same goes for the APY displayed in the vesting page)

image

**Review by tracyarciaga — COMMENTED** [review]
@barryhalo can we format this APY column to include a comma? (same goes for the APY displayed in the vesting page)

So far, code looks correct based on Chris' computation.

image

**Comment by barryhalo** [comment]
> @barryhalo can we format this APY column to include a comma? (same goes for the APY displayed in the vesting page)
> 
> So far, code looks correct based on Chris' computation.
> 
> image

Okay on it @tracyarciaga

File: src/components/Farm/FarmPoolCard.tsx (hunk 1 above)
Comment (chris247474) [reply src/components/Farm/FarmPoolCard.tsx]: @barryhalo sounds good

**Comment by barryhalo** [comment]
@tracyarciaga APY comma has been fixed

Screen Shot 2021-06-07 at 6 40 25 PM

@schystz 
Added HALO token in `COINGECKO_KNOWN_TOKENS`

File: src/components/Farm/FarmPoolCard.tsx [hunk 2]
```diff
@@ -403,22 +404,37 @@ export default function FarmPoolCard({ poolInfo, tokenPrice }: FarmPoolCardProps
...
+  const rewardTokenPerSecond = useRewardTokenPerSecond()
+
+  // HALO address
+  const HALOToken = process.env.REACT_APP_HALO_REWARDS_ADDRESS_KOVAN
+    ? process.env.REACT_APP_HALO_REWARDS_ADDRESS_KOVAN
+    : ''
+
+  // 1 month in seconds
+  // (days * hrs * min * s) * reward token per second
+  const monthlyReward = rewardTokenPerSecond ? 30 * 24 * 60 * 60 * rewardTokenPerSecond : 0
```
Comment (chris247474) [inline src/components/Farm/FarmPoolCard.tsx]: @barryhalo looks great! thoughts on putting this in a helper func? that way any unit tests can just import it and test against it.

**Review by chris247474 — CHANGES_REQUESTED** [review]
@barryhalo equation seems to make sense (that's all i checked, @schystz would be a better check on the rest of the code :) ). Just left a comment on putting the logic in a helper file. 

Is the unit test for the APY calculation in this PR as well, sorry if i missed that

File: src/components/Farm/FarmPoolCard.tsx (hunk 2 above)
Comment (barryhalo) [reply src/components/Farm/FarmPoolCard.tsx]: Yeah good catch, on it!

Comment (barryhalo) [reply src/components/Farm/FarmPoolCard.tsx]: Ahh you mean the whole APY computation, got it!

Comment (barryhalo) [reply src/components/Farm/FarmPoolCard.tsx]: @ctverceles I didn't include the unit tests here, created a new task for that
https://halodao.atlassian.net/browse/HDF-305

Comment (barryhalo) [reply src/components/Farm/FarmPoolCard.tsx]: @ctverceles I guess I can include the unit tests here, and just link its Jira issue :)

File: src/components/Farm/FarmPoolCard.tsx [hunk 3]
```diff
@@ -403,22 +404,37 @@ export default function FarmPoolCard({ poolInfo, tokenPrice }: FarmPoolCardProps
...
 
-  const [claimRewards] = useClaimRewardsCallback()
+  // Pool Liquidity
+  const poolLiquidity = getPoolLiquidity(poolInfo, tokenPrice)
+
+  // APY
+  const rewardTokenPerSecond = useRewardTokenPerSecond()
+
+  // HALO address
+  const HALOToken = process.env.REACT_APP_HALO_REWARDS_ADDRESS_KOVAN
```
Comment (ernestpascual) [inline src/components/Farm/FarmPoolCard.tsx]: i think we can use the `HALO_TOKEN_ADDRESS` here from `src/constants/index` this should change the address during network changes. This applies to halohalo and reward addresses as well

File: src/components/Farm/FarmPoolCard.tsx (hunk 2 above)
Comment (ernestpascual) [reply src/components/Farm/FarmPoolCard.tsx]: We can also use a custom hook actually here

File: src/constants/index.ts [hunk 4]
```diff
@@ -122,6 +122,10 @@ export const BALANCER_SUBGRAPH_URL =
   process.env.REACT_APP_BALANCER_SUBGRAPH_URL ||
   'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer/graphql'
 
+const HALOToken = process.env.REACT_APP_HALO_REWARDS_ADDRESS_KOVAN
```
Comment (ernestpascual) [inline src/constants/index.ts]: For this i think u can use line 104 instead?

**Review by ernestpascual — CHANGES_REQUESTED** [review]
added with some comments. also noticed there's a bunch of eslint things here as well

File: src/components/Farm/FarmPoolCard.tsx (hunk 3 above)
Comment (barryhalo) [reply src/components/Farm/FarmPoolCard.tsx]: > i think we can use the `HALO_TOKEN_ADDRESS` here from `src/constants/index` this should change the address during network changes. This applies to halohalo and reward addresses as well

@ernestpascual 

Resolved

File: src/constants/index.ts (hunk 4 above)
Comment (barryhalo) [reply src/constants/index.ts]: > For this i think u can use line 104 instead?

@ernestpascual 

Resolved

**Comment by barryhalo** [comment]
@ernestpascual @schystz 

APY hook is ready for review

**Comment by schystz** [comment]
Closing this PR to submit a fresh one. It looks like GH is a little lost as this PR is showing 29 files changed while if you try to open a new PR for the same changes (https://github.com/HaloDAO/halodao-interface/compare/develop...feature/hdf-13-apy) there are only 9 files changed.

**Pull request closed without merging**